src/org/traccar/protocol/TkmateProtocolDecoder.java

```diff
@@ -0,0 +1,239 @@
+            .number("(dd)(dd)(dd)|")              // date
+            .number("(d+.d+)|")                  // speed
+            .number("(d+)|")                     // heading
+            .number("(d+)|")                     // ignition
+            .number("(d+)|")                     // dop1
+            .number("(d+)|")                     // dop2
+            .number("(d+.d+)|")                  // analog
+            .number("(d+.d+)|")                  // internal battery
+            .number("(d+.d+)|")                  // vehicle battery
+            .number("(d+.d+)|")                  // gps odometer
+            .number("(d+.d+)|")                  // pulse odometer
+            .number("(d+)|")                     // main power status
+            .number("(d+)|")                     // gps data validity
+            .number("(d+)|")                     // live or cache
+            .any()
+            .compile();
+
+
+    private static final Pattern PATTERN_ALT = new PatternBuilder()
+            .text("^TMALT|")
+            .expression("([^ ]+)|")              // uid
+            .number("(d+)|")                     // seq
+            .number("(d+)|")                     // Alert type
+            .number("(d+)|")                     // Alert status
+            .number("(d+.d+)|")                  // latitude
+            .number("(d+.d+)|")                  // longitude
+            .number("(dd)(dd)(dd)|")              // time
+            .number("(dd)(dd)(dd)|")              // date
+            .number("(d+.d+)|")                  // speed
+            .number("(d+)|")                     // heading
+            .any()
+            .compile();
+
+
+    private String decodeAlarm(int value) {
+        switch (value) {
+            case 1:
+                return Position.ALARM_SOS;
+            case 2:
+                return null;
+            case 3:
+                return Position.ALARM_GEOFENCE;
+            case 4:
+                return Position.ALARM_POWER_CUT;
+            case 5:
+                return null;
+            case 6:
+                return null;
+            default:
+                return null;
+        }
+    }
+    private Object decodeSrt(Channel channel, SocketAddress remoteAddress, String sentence) {
+        Parser parser = new Parser(PATTERN_SRT, sentence);
+        if (!parser.matches()) {
+            return null;
+        }
+
+        Position position = new Position();
+        position.setProtocol(getProtocolName());
+        DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, parser.next());
+        if (deviceSession == null) {
+            return null;
+        }
+        position.setDeviceId(deviceSession.getDeviceId());
+
+        position.setLatitude(parser.nextDouble());
+        position.setLongitude(parser.nextDouble());
+
+        DateBuilder dateBuilder = new DateBuilder(TimeZone.getTimeZone("Asia/Calcutta"))
```

Comment:
While testing without timezone i have found that timezone(which is gmt+5.30) is being added to time field of packet. For example, ^TMSRT|354678456723764|12.59675|77.56789|**123456**|030414|1.03|1.01|#. After parsing this packet, time becomes 18:04:56.Hence I used particular timezone to avoid this problem. Please advice right approach to solve this problem.